cpp/src/arrow/io/interfaces.cc

```diff
@@ -88,14 +91,6 @@ Result<Iterator<std::shared_ptr<Buffer>>> MakeInputStreamIterator(
   return Iterator<std::shared_ptr<Buffer>>(InputStreamBlockIterator(stream, block_size));
 }
 
-Status Readable::Read(int64_t nbytes, int64_t* bytes_read, void* out) {
```

Comment:
Hmm... why are you removing those?

Reply:
Huh. I think this is a merge error with your io -> result PR. I'll revert it.

Maybe we should have a `deprecated_test.cc` somewhere to ensure that we at least get a link error if this happens again?